### DISCUSSION TITLE
Add a way to override `url.origin` when running nextjs app locally through a proxy like ngrok

### DISCUSSION THREAD

**Discussion by spennyp** [wevm/frog-archived#75] [Ideas]
It is useful to run a frame server locally, and expose it via a proxy like ngrok to debug frames inside of Warpcast. 

When running a nextjs app locally and using a proxy, `url.origin` will be `localhost`, which breaks the frame when used through the proxy. It would be ideal to have a way to override `url.origin` to use the proxy address. 

https://github.com/wevm/frog/blob/bd4c7740f7497d9b146db4ca63e828af8ed448ca/src/frog-base.tsx#L268-L270

**Comment by ggomaeng** [comment]
You can pass `origin` to the frog options https://frog.fm/reference/frog#origin

**Reply by filipesoccol** [reply]
Using `origin` doesn't work for me behind a NGINX proxy. For some reason it turns back to "127.0.0.1" for images and redirection paths.

I've tested on the default node and bun templates. Both generate image files correctly on the domain, but the frog paths to images and routes are wrong. As I've reported here: https://discord.com/channels/1156791276818157609/1212219334370529370
